file-service-it/src/test/java/uk/gov/justice/services/fileservice/it/FileServiceIntegrationTest.java

```diff
@@ -0,0 +1,112 @@
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+import uk.gov.justice.services.fileservice.api.FileRetriever;
+import uk.gov.justice.services.fileservice.api.FileServiceException;
+import uk.gov.justice.services.fileservice.api.FileStorer;
+import uk.gov.justice.services.fileservice.client.FileService;
+import uk.gov.justice.services.fileservice.domain.FileReference;
+import uk.gov.justice.services.fileservice.it.helpers.IntegrationTestConnectionProvider;
+import uk.gov.justice.services.fileservice.json.StringJsonSetter;
+import uk.gov.justice.services.fileservice.repository.ContentJdbcRepository;
+import uk.gov.justice.services.fileservice.repository.FileStore;
+import uk.gov.justice.services.fileservice.repository.MetadataJdbcRepository;
+import uk.gov.justice.services.fileservice.it.helpers.LiquibaseDatabaseBootstrapper;
+import uk.gov.justice.services.fileservice.util.Closer;
+import uk.gov.justice.services.fileservice.util.NewIdProvider;
+import uk.gov.justice.services.jdbc.persistence.InitialContextFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.UUID;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.json.JsonObject;
+import javax.sql.DataSource;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.openejb.jee.Application;
+import org.apache.openejb.jee.WebApp;
+import org.apache.openejb.junit.ApplicationComposer;
+import org.apache.openejb.testing.Classes;
+import org.apache.openejb.testing.Module;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(ApplicationComposer.class)
+public class FileServiceIntegrationTest {
+
+    private static final String LIQUIBASE_FILE_STORE_DB_CHANGELOG_XML = "liquibase/file-service-liquibase-db-changelog.xml";
+
+    private final LiquibaseDatabaseBootstrapper liquibaseDatabaseBootstrapper = new LiquibaseDatabaseBootstrapper();
+
+    @Resource(name = "openejb/Resource/DS.fileservice")
+    private DataSource dataSource;
+
+    @Inject
+    FileService fileService;
+
+    @Module
+    @Classes(cdi = true, value = {
+
+            FileRetriever.class,
+            FileServiceException.class,
+            FileStorer.class,
+            FileReference.class,
+
+            InitialContextFactory.class,
+
+            FileService.class,
+
+            IntegrationTestConnectionProvider.class,
+
+            StringJsonSetter.class,
+            ContentJdbcRepository.class,
+            FileStore.class,
+            MetadataJdbcRepository.class,
+
+            Closer.class,
+            NewIdProvider.class
+
+    })
+    public WebApp war() {
+        return new WebApp()
+                .contextRoot("core-test")
+                .addServlet("TestApp", Application.class.getName());
+    }
+
+    @Before
+    public void initDatabase() throws Exception {
```

Comment:
Can we do this once per class rather than once per test? Ideally, we would be able to run our ITs in parallel so we wouldn't be able to rely on having a clean DB for every test.